src/ophyd_async/epics/pvi/pvi.py

```diff
@@ -179,7 +179,15 @@ def _mock_common_blocks(device: Device, stripped_type: Optional[Type] = None):
                 sub_device_2 = device_cls(SoftSignalBackend(signal_dtype))
                 sub_device = DeviceVector({1: sub_device_1, 2: sub_device_2})
             else:
-                sub_device = DeviceVector({1: device_cls(), 2: device_cls()})
+                if hasattr(device, device_name):
+                    continue
+                else:
+                    sub_device = DeviceVector(
+                        {
+                            1: getattr(device, device_name, device_cls()),
+                            2: getattr(device, device_name, device_cls()),
+                        }
+                    )
 
                 for sub_device_in_vector in sub_device.values():
                     _mock_common_blocks(sub_device_in_vector, stripped_type=device_cls)
```

Comment:
You've already checked if `device` has the `device_name` attr.

We should also still recurse on this if if pre-declare-blocks was called:

```suggestion
                if hasattr(device, device_name):
                    sub_device = getattr(device, device_name)
                else:
                    sub_device = DeviceVector(
                        {
                            1: device_cls(),
                            2: device_cls()
                        }
                    )

                for sub_device_in_vector in sub_device.values():
                    _mock_common_blocks(sub_device_in_vector, stripped_type=device_cls)
```